sympy/integrals/transforms.py

```diff
@@ -1150,7 +1155,75 @@ class LaplaceTransform(IntegralTransform):
 
     _name = 'Laplace'
 
+    def _cr(self, f, a, c, **hints):
+        conds = not hints.get('noconds', False)
+        if conds:
+            return f, a, c
+        else:
+            return f
+
     def _compute_transform(self, f, t, s, **hints):
+        a = Wild('a', exclude=[t])
+        b = Wild('b', exclude=[t])
+        x = Wild('x')
+        y = Wild('y')
+        tau = Wild('tau', exclude=[t])
+        omega = Wild('omega', exclude=[t])
+        # Expressions with exponential functions:
+        # http://eqworld.ipmnet.ru/en/auxiliary/inttrans/laplace3.pdf
+        # 3.1
+        ma1 = f.match(b*exp(x))
+        if ma1:
+            ma2 = ma1[x].collect(t).match(a*t-tau)
+            if ma2:
+                return self._cr(ma1[b]*exp(-ma2[tau])/(s-ma2[a]),
+                                ma2[a], S.true, **hints)
```

Comment:
This block of code seems to be repeated many times. Maybe we can just collect all the rules together like:
```python
rules = [
    (b*exp(a*t - tau), b*exp(-tau)/(s - a)),
    ... 
]
```
Then the main code can just do something like:
```python
for pattern, replacement in rules:
     if f.match(pattern)
         ...
```
Also maybe leading constants can be factored out of the pattern at the beginning so it's something like:
```python
for term in Add.make_args(f):
    coeff, func = f.as_independent(t, as_Add=False)
    # Now laplace transform func
```

Reply:
I tried that first, but matching to `b*exp(a*t - tau)` will not necessarily work, which is why I do a `collect(t)` on `ma1[x]`. Sometimes I also have to check parameter values, I will mark an example.